scipy/optimize/tests/test_chandrupatla.py

```diff
@@ -769,13 +779,14 @@ def f(x):
         assert_allclose(res.x, 1)
 
         # Test that if both ends of bracket equal root, algorithm reports
-        # convergence
-        def f(x):
-            return x**2 - 1
+        # convergence.
+        def f(x, root):
+            return x**2 - root
```

Comment:
Adjust the function so `root` is always the root (not just for the values of `root` tested).
```suggestion
        def f(x, root):
            return (x - root)**2
```